docs/api/toolkits.md

```diff
@@ -0,0 +1,112 @@
+# Toolkits Wrappers and Registries
+
+The toolkit wrappers provide a simple uniform API for accessing minimal functionality of cheminformatics toolkits.
+
+These toolkit wrappers are generally used through a {class}`ToolkitRegistry`, which can be constructed with a desired precedence of toolkits:
+
+```python
+>>> from openff.toolkit.utils.toolkits import ToolkitRegistry, OpenEyeToolkitWrapper, RDKitToolkitWrapper, AmberToolsToolkitWrapper
+>>> toolkit_registry = ToolkitRegistry()
+>>> toolkit_precedence = [OpenEyeToolkitWrapper, RDKitToolkitWrapper, AmberToolsToolkitWrapper]
+>>> [ toolkit_registry.register_toolkit(toolkit) for toolkit in toolkit_precedence if toolkit.is_available() ]
+[None, None, None]
+
+```
+
+The toolkit wrappers can then be accessed through the registry:
+
+```python
+>>> from openff.toolkit.utils.toolkits import GLOBAL_TOOLKIT_REGISTRY as toolkit_registry
+>>> from openff.toolkit import Molecule
+>>> molecule = Molecule.from_smiles('Cc1ccccc1')
+>>> smiles = toolkit_registry.call('to_smiles', molecule)
+
+```
+
+The order of toolkits, as specified in `toolkit_precedence` above, determines the order in which
+the called method is resolved, i.e. if the toolkit with highest precedence has a `to_smiles`
+method, that is the toolkit that will be called. If the toolkit with highest precedence does not
+have such a method, it is attempted with other toolkits until one is found. By default, if a toolkit with an appropriately-named method raises an exception of any type, then iteration over the registered toolkits stops and that exception is raised. To continue iteration if specific exceptions are encountered, customize this behavior using the optional `raise_exception_types` keyword argument to `ToolkitRegistry.call`. If no registered
+toolkits have the method, a ValueError is raised, containing a message listing the registered toolkits and exceptions (if any) that were ignored. 
```

Comment:
I had a very long comment written that got eaten by my browser ... I'll try to summarize it.

I suggest this be restructure a little bit, drawing on what is already here. I'd like to see something that communicates, mostly in this order

* How a registry selects wrappers to try
* What happens when the wrapper runs a method without error
* What happens when the (first attempted) wrapper throws an error
* How to raise specific errors but skip others
* What happens when no wrappers have a matching method name
* Example showing default behavior of something working
* Example showing default behavior of a wrapper throwing an error
* Example showing how `raise_exception_types` can be used to skip one type of error but throw another one

This is fairly long, but I think this paragraph currently tries to convey much more information that can fit into one section of text. I find this behavior of registries and wrappers pretty confusing [^1]- my suggestion here is not a criticism of the prose as-is but my attempt to project backwards in time what I wish existed in the docs three years ago.

> To continue iteration if specific exceptions are encountered, customize this behavior using the optional `raise_exception_types` keyword argument to `ToolkitRegistry.call`.

If I understand the point of this Markdown file, this detail needs to be expanded on (even if the source code already exists as a source of truth). I had to look at the docstrings to remember how I would use this argument to do that.

[^1]: For example, `raise_exception_types` is meant to be of type `list[Exception]` but its default value is `None`. In that case, it's transformed into `list[Exception]`, and in English that translates to something like "raise only these specific exception types, but oh by the way if you don't pick any I will assume you meant 'all of them'."